File: src/main/java/axirassa/webapp/pages/monitor/StatisticsMonitor.java

package axirassa.webapp.pages.monitor;

import org.apache.shiro.authz.annotation.RequiresUser;
import org.apache.tapestry5.annotations.Property;
import org.apache.tapestry5.ioc.annotations.Inject;
import org.hibernate.Session;

import axirassa.dao.PingerDAO;
import axirassa.model.PingerEntity;

@RequiresUser
public class StatisticsMonitor {
	@Inject
	private Session session;

	@Inject
	private PingerDAO pingerDAO;

	@Property
	private PingerEntity pinger;

	@Property
	private Long id;


	public void onActivate(Long id) {
		this.id = id;

		pinger = pingerDAO.findPingerById(id);
	}
}

File: src/main/java/axirassa/webapp/pages/monitor/StatisticsMonitor.scala
package axirassa.webapp.pages.monitor

import axirassa.webapp.services.AxirassaSecurityException
import axirassa.model.PingerEntity
import axirassa.model.HttpStatisticsEntity
import org.apache.tapestry5.annotations.Property
import axirassa.dao.PingerDAO
import org.apache.tapestry5.services.Session
import org.apache.tapestry5.ioc.annotations.Inject
import org.apache.shiro.authz.annotation.RequiresUser

@RequiresUser
class StatisticsMonitor {
    @Inject
    var session : Session = _

    @Inject
    var pingerDAO : PingerDAO = _

    @Property
    var statistics : List[HttpStatisticsEntity] = _

    @Property
    var statistic : HttpStatisticsEntity = _

    @Property
    var pinger : PingerEntity = _

    @Property
    var id : Long = _

    def onActivate (id: Long) {
        this.id = id

        val pingerById = pingerDAO.findPingerById(id)        
        if(pingerById.isEmpty)
            throw new AxirassaSecurityException
            
        pinger = pingerById.get
        statistics = pingerDAO.findStatistics(pinger)
    }
}